src/looprestoration_tmpl_16.rs

```diff
@@ -1034,13 +1018,13 @@ unsafe extern "C" fn loop_restoration_dsp_init_x86(
         return;
     }
 
-    (*c).wiener[0] = Some(dav1d_wiener_filter7_16bpc_ssse3);
-    (*c).wiener[1] = Some(dav1d_wiener_filter5_16bpc_ssse3);
+    (*c).wiener[0] = LoopRestorationFilterFn::Bpc16(dav1d_wiener_filter7_16bpc_ssse3);
```

Comment:
Oh, you did it this way.  I was thinking of having something like `LoopRestorationFilterFn::BPC16_SSSE3` so there are no more `fn` ptrs at all.  Doing it this way, we'd have both the `enum` `match` and the indirect `fn` ptr call and we don't get the other benefits from entirely removing the `fn` ptrs.

Reply:
Oh, that's a good point. Let me rework this to do away with fn pointers.